e2e/cypress/e2e/settings/features.cy.ts

```diff
@@ -0,0 +1,216 @@
+
+      // Check that feature toggles are present
+      cy.get('cnsl-feature-toggle').should('have.length.greaterThan', 0);
+      cy.get('cnsl-login-v2-feature-toggle').should('be.visible');
+    });
+
+    it('should maintain feature states after page reload', () => {
+      // Wait for features to load
+      cy.get('cnsl-feature-toggle').should('be.visible');
+
+      // Get the count of feature toggles before reload
+      cy.get('cnsl-feature-toggle').then(($toggles) => {
+        const initialCount = $toggles.length;
+
+        // Reload page
+        cy.reload();
+
+        // Wait for features to load again
+        cy.get('cnsl-feature-toggle').should('be.visible');
+
+        // Verify that the same number of toggles are still present
+        cy.get('cnsl-feature-toggle').should('have.length', initialCount);
+
+        // Verify that toggles are still functional
+        cy.get('cnsl-feature-toggle')
+          .first()
+          .within(() => {
+            cy.get('mat-button-toggle').should('be.visible');
+          });
+      });
+    });
+
+    it('should handle API errors gracefully', () => {
+      // Intercept API calls and force them to fail
+      cy.intercept('POST', '**/features*', {
+        statusCode: 500,
+        body: { message: 'Internal server error' },
+      }).as('featureError');
+
+      // Try to toggle a feature
+      cy.get('cnsl-feature-toggle')
+        .first()
+        .within(() => {
+          cy.get('mat-button-toggle').first().click();
+        });
+
+      // Wait for the error response (optional since it might not always trigger)
+      cy.get('body').then(() => {
+        // Just verify that the page is still functional
+        cy.get('cnsl-feature-toggle').should('be.visible');
+      });
+    });
+
+    it('should be keyboard accessible', () => {
+      // Features should be focusable via the underlying button
+      cy.get('cnsl-feature-toggle')
+        .first()
+        .within(() => {
+          cy.get('mat-button-toggle button').first().focus().should('be.focused');
+        });
+
+      // Reset button should be focusable if it exists
+      cy.get('body').then(($body) => {
+        if ($body.find('button:contains("Reset")').length > 0) {
+          cy.get('button').contains('Reset').focus().should('be.focused');
+        }
+      });
+    });
+
+    describe('permissions', () => {
+      it('should show appropriate elements for admin users', () => {
+        // Admin should see reset button if they have permissions
+        cy.get('body').then(($body) => {
+          if ($body.find('button:contains("Reset")').length > 0) {
+            cy.get('button').contains('Reset').should('be.visible');
+          }
+        });
+
+        // Admin should see all feature toggles
+        cy.get('cnsl-feature-toggle').should('have.length.greaterThan', 0);
+      });
+    });
+  });
+
+  describe('Feature Modification Tests', () => {
+    afterEach(() => {
```

Comment:
Also beforeEach so a failing modification test can be fixed rerun with a clean state?